Source/Orts.Simulation/Simulation/RollingStocks/MSTSSteamLocomotive.cs

```diff
@@ -5223,7 +5194,7 @@ protected override void UpdateTractiveForce(float elapsedClockSeconds, float t,
                     float connectRodInertiaForcelbf = inertiaSpeedCorrectionFactor * connectRodInertiaAngleFactor * ConnectingRodWeightLb;
 
                     // Account for the position of the crosshead position. In other words it depends upon whether the Rods and Reciporating gear is above or below the axle.
-                    if (crankAngleRad > 0 && crankAngleRad < Math.PI)
+                    if (crankAngleRad > 0 && crankAngleRad < Math.PI && normalisedCrankAngleRad < Math.PI)
```

Comment:
I don't understand why you are using crankAngleRad and normalisedCrankAngleRad. You should use only normalisedCrankAngleRad, which I'd suggest that you rename to crankAngleRad, i.e.
```csharp
float crankAngleRad = NormalisedCrankAngle(i);
```

Reply:
For force calculation a reference angle of 0 -180 - 0 angle is used, this lines up with the result tables in the reference book that I got the feature description from (and is also what was used in the confirmation spreadsheet that we have been circulating).

To confirm whether the locomotive cylinder is on the forward or backward stroke the normalised crank angle is used. It is also used to calculate the opening and closing points for the cylinder exhausts.

Hence my preference is to maintain both of them.

Reply:
I find really confusing that some of the formulas range from -180º to 180º and others go from 0º to 360º. I strongly suggest that you rewrite everything to use the same convention. Otherwise, other people won't understand what you are doing, and it's going to be bug prone.

In particular, the following line is terribly confusing, and has to be removed.
```csharp
if (crankAngleRad > 0 && crankAngleRad < Math.PI && normalisedCrankAngleRad < Math.PI)
```
It would be enough to test if (crankAngleRad > 0), since the rest of conditions are equivalent to that.

Reply:
I think that the crank angle range should remain in the range 0 - 180 - 0 so that it matches the reference data tables and makes it easier to compare to actual results in OR. This makes debugging easier.

I have added some additional notes to try and make it easier to understand what the different angle ranges are used for, etc.

I have simplified some of the code based upon your suggested feedback.

Hopefully this is an agreeable compromise.